src/librustc_trait_selection/traits/error_reporting/suggestions.rs

```diff
@@ -218,51 +218,29 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
                     return;
                 }
 
-                hir::Node::Item(hir::Item {
-                    kind: hir::ItemKind::Struct(_, generics),
-                    span,
-                    ..
-                })
-                | hir::Node::Item(hir::Item {
-                    kind: hir::ItemKind::Enum(_, generics), span, ..
-                })
-                | hir::Node::Item(hir::Item {
-                    kind: hir::ItemKind::Union(_, generics),
-                    span,
-                    ..
-                })
+                hir::Node::Item(hir::Item { kind: hir::ItemKind::Struct(_, generics), .. })
+                | hir::Node::Item(hir::Item { kind: hir::ItemKind::Enum(_, generics), .. })
+                | hir::Node::Item(hir::Item { kind: hir::ItemKind::Union(_, generics), .. })
                 | hir::Node::Item(hir::Item {
                     kind: hir::ItemKind::Trait(_, _, generics, ..),
-                    span,
-                    ..
-                })
-                | hir::Node::Item(hir::Item {
-                    kind: hir::ItemKind::Impl { generics, .. },
-                    span,
                     ..
                 })
                 | hir::Node::Item(hir::Item {
-                    kind: hir::ItemKind::Fn(_, generics, _),
-                    span,
-                    ..
+                    kind: hir::ItemKind::Impl { generics, .. }, ..
                 })
+                | hir::Node::Item(hir::Item { kind: hir::ItemKind::Fn(_, generics, _), .. })
                 | hir::Node::Item(hir::Item {
-                    kind: hir::ItemKind::TyAlias(_, generics),
-                    span,
-                    ..
+                    kind: hir::ItemKind::TyAlias(_, generics), ..
                 })
                 | hir::Node::Item(hir::Item {
-                    kind: hir::ItemKind::TraitAlias(generics, _),
-                    span,
-                    ..
+                    kind: hir::ItemKind::TraitAlias(generics, _), ..
                 })
                 | hir::Node::Item(hir::Item {
                     kind: hir::ItemKind::OpaqueTy(hir::OpaqueTy { generics, .. }),
-                    span,
                     ..
                 })
-                | hir::Node::TraitItem(hir::TraitItem { generics, span, .. })
-                | hir::Node::ImplItem(hir::ImplItem { generics, span, .. })
+                | hir::Node::TraitItem(hir::TraitItem { generics, .. })
+                | hir::Node::ImplItem(hir::ImplItem { generics, .. })
```

Comment:
Let's simplify this further with `#![feature(or_patterns)]` by moving the `|`s as much inwards as possible.